src/test/rustdoc/keyword.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![crate_name = "foo"]
+
+// @has foo/index.html '//h2[@id="keywords"]' 'Keywords'
+// @has foo/index.html '//a[@href="keyword.match.html"]' 'match'
+// @has foo/keyword.match.html '//a[@class="keyword"]' 'match'
+// @has foo/keyword.match.html '//section[@id="main"]//div[@class="docblock"]//p' 'this is a test!'
+#[doc(keyword = "match")]
+/// this is a test!
+mod foo{}
```

Comment:
Should we also include a test here to make sure the `foo/foo/index.html` and related links are missing? Seems like a good chance to use the `@!has_dir` command.

Reply:
I was hoping to have this checked directly once we have added keywords into std docs directly.

Reply:
...wait, how would that work? Isn't it the job of these tests to make sure the feature works, without depending on how it's used to pull that together?

Reply:
What I meant is that if it fails to link once we use the keyword feature in std docs, it'll fail tests at this moment. :)

Reply:
I'm not worried about it placing broken links in the docs - the test is already checking for that. I'm worried about rustdoc creating useless unlinked pages for the original module that's supposed to not exist once rustdoc processes the keyword information out of it. (To be sure, since this code was adapted from the code for primitives, this issue shouldn't exist, but I like more tests.)

What I am suggesting is adding lines like this:

```
// @!has foo/index.html '//a/@href' 'foo/index.html'
// @!has foo/foo/index.html
```